dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -288,7 +290,7 @@ public Set<String> getOrderableFields() {
   }
 
   private Predicate<TrackedEntity> filterAccessibleTrackedEntities(
-      UserDetails user, TrackedEntityQueryParams queryParams) {
+      UserDetails user, TrackedEntityQueryParams queryParams, List<Program> programs) {
     boolean skipOwnershipCheck = queryParams.getOrgUnitMode() == ALL;
 
     if (queryParams.hasEnrolledInTrackerProgram()) {
```

Comment:
Not part of this PR but this logic could be simplified with something like:
`return te -> (skipOwnershipCheck ||
ownershipAccessManager.hasAccess(user, te, queryParams.getEnrolledInTrackerProgram());`